contracts/Verifier.sol

```diff
@@ -197,47 +197,23 @@ contract Verifier is EVMConstants, Ownable {
             return;
         }
 
-        uint pc = _executionState.pc;
-        uint codeptr = 0;
-        uint codeSize = 0;
-        address codeContractAddress = dispute.codeContractAddress;
-
-        // TODO: what happens if the code make assumptions with OP.PC ?
-        //       should we pad the code array with zeros?
-        //       -
-        //       we can do some prechecks without running the evm for:
-        //       - JUMP, JUMPI
-        //       - CODESIZE
-        //       - maybe others too
-        if (_executionState.isCodeCompacted) {
-            uint pcEnd = _executionState.pcEnd;
-
-            if (pc == pcEnd) {
-                pcEnd += 1;
-            }
+        if ((dispute.state & END_OF_EXECUTION) != 0) {
+            address codeAddress = dispute.codeContractAddress;
+            uint pos = _executionState.pc;
+            uint8 opcode;
 
-            codeSize = pcEnd - pc;
-            codeptr = pc;
-            _executionState.pc = 0;
-        } else {
             assembly {
-                codeSize := extcodesize(codeContractAddress)
+                extcodecopy(codeAddress, 31, pos, 1)
+                opcode := mload(0)
             }
-        }
 
-        bytes memory code = new bytes(codeSize);
-        assembly {
-            extcodecopy(codeContractAddress, add(code, 0x20), codeptr, codeSize)
-        }
-
-        if ((dispute.state & END_OF_EXECUTION) != 0) {
-            if (uint8(code[_executionState.pc]) != OP_REVERT && uint8(code[_executionState.pc]) != OP_RETURN) {
+            if (opcode != OP_REVERT && opcode != OP_RETURN) {
```

Comment:
This question is unrelated to the PR.
Now I think more about this, I wonder if the operator in plasma chain would submit reverted transactions to main net?
Is this will open a kind of DOS attack?

Reply:
This can be challenged. This just makes sure that a challenger can proof that the solvers solution was too 'long'

Reply:
I see, this is just for the case solver didn't stop after hitting REVERT. My question was totally unrelated.
But given that, then stateHash is already enough to cover this case, isn't it?
Because if opcode is REVERT, EVMRuntime will return some error?

Reply:
Also there is opcode `invalid`. Should it be included here?

Reply:
Yes, stateHash is enough because they have to proof the execution and also agree on prioer execution steps.

>> Also there is opcode invalid. Should it be included here?

Hmm, no. Every valid execution ends with either RETURN or REVERT. 
Or most of the time: Out Of Gas xD

Reply:
I think so too. When something invalid happen it will REVERT.
But here: https://solidity.readthedocs.io/en/v0.5.2/assembly.html#opcodes
there is an `invalid` which also said to `end execution with invalid instruction`
Am I miss anything...

Reply:
I think `assert` will end with `invalid` but haven't found something conclusive

Reply:
@peara 
If we want to allow this, we might also include SELFDESTRUCT. But that wasn't at the table.
I think we don't want to allow that for PLASMA @johannbarbie ?

Reply:
@peara But still pointer, STOP is missing 😎